gradle-plugin/src/main/kotlin/co/touchlab/kotlin/gradle/targets/native/cocoapods/CocoapodsExtension.kt

```diff
@@ -47,17 +48,27 @@ open class CocoapodsExtension(private val project: Project) {
     @Input
     var homepage: String? = null
 
-    /**
-     * Configure framework name of the pod built from this project.
-     */
-    @Input
-    var frameworkName: String = project.name.asValidFrameworkName()
+    @Internal
+    internal var frameworkName: String = project.name.asValidFrameworkName()
+    @Internal
+    internal var isStatic: Boolean = true
+    @Internal
+    internal var frameworkConfiguration: Framework.() -> Unit = {
+        baseName = project.name.asValidFrameworkName()
+        isStatic = true
+    }
 
-    /**
-     * Configure if framework should be static.
-     */
+    internal fun configureFramework(framework: Framework){
```

Comment:
I dislike how all this is set up, but it seemed to be the only way to match the syntax and capabilities of configuring the binary while also making certain settings available to be used by the tasks in cocoapodTasks

I suspect there is a much tidier way of doing this which is just eluding me

Reply:
I guess another way to do it might be to cache the `Framework` object and expose `frameworkName` and `isStatic` as vals with custom getters, instead of vars. But I don't know that it makes a huge difference. You could add `private set` to them if you want to avoid them being mutated outside the extension.

Reply:
Pushed an update which I think address both your comments. Rather than having an internal copy of the configs we care about we get it from the kotlin extension when we need it

Reply:
I mean, I think our fork is experimental, so there's no value in keeping "isStatic" on the config if we're going to focus on direct Framework config

Reply:
Disregard. I was seeing something wrong.